src/Files.App/Helpers/Win32/Win32Helper.Storage.cs

```diff
@@ -935,10 +943,72 @@ public static bool UnsetFileAttribute(string lpFileName, FileAttributes dwAttrs)
 			return Win32PInvoke.SetFileAttributesFromApp(lpFileName, lpFileInfo.dwFileAttributes & ~dwAttrs);
 		}
 
+		public static unsafe bool CanCompressContent(string path)
+		{
+			path = Path.GetPathRoot(path) ?? string.Empty;
+
+			// MAX_PATH = 260 + 1 (null termination)
+			var volumeName = new PWSTR();
+			var fileSystemName = new PWSTR();
```

Comment:
Don't you have to allocate? If so, you can do as follows:

```c#
const int MAX_PATH = 260;
PWTSR pVolumeName = stackalloc char[MAX_PATH]; // you may need casting maybe?
PWSTR pFileSystemName = stackalloc char[MAX_PATH];
```

Reply:
In that cause, I think you can specify null to the dedicated parameter.

```c#
PInvoke.GetVolumeInformation(
    path,
    null,
    0u,
    null,
    null,
    &dwFileSystemFlags,
    null,
    0u);
```

Reply:
Addressed in [`c50e313` (#16319)](https://github.com/files-community/Files/pull/16319/commits/c50e31335f13669d3619e4de79a20032a537160b)